app/assets/javascripts/components/place_autocomplete.js

```diff
@@ -36,6 +36,24 @@
         $(_this.opts.fillCity).val("")
       }
 
+      var countyComponent = place.address_components.find(function(c) {
+        return c.types.indexOf("administrative_area_level_2") >= 0
+      })
+      if (!!countyComponent) {
+        $(_this.opts.fillCounty).val(countyComponent.long_name)
+      } else {
+        $(_this.opts.fillCounty).val("")
+      }
+
+      var stateComponent = place.address_components.find(function(c) {
+        return c.types.indexOf("administrative_area_level_1") >= 0
+      })
+      if (!!stateComponent) {
+        $(_this.opts.fillState).val(stateComponent.short_name)
+      } else {
+        $(_this.opts.fillState).val("")
+      }
```

Comment:
any suggestion to refactor these similar logic?

Reply:
I think it's simple enough to not worry about the duplications